api/v1alpha1/oadp_types.go

```diff
@@ -110,6 +110,12 @@ type VeleroConfig struct {
 	// Default is 10m
 	// +optional
 	ResourceTimeout string `json:"resourceTimeout,omitempty"`
+	// Maximum number of requests by the server to the Kubernetes API in a short period of time. (default 30)
+	// +optional
+	ClientBurst *int `json:"client-burst,omitempty"`
+	// Maximum number of requests per second by the server to the Kubernetes API once the burst limit has been reached. (default 20)
+	// +optional
+	ClientQPS *float32 `json:"client-qps,omitempty"`
```

Comment:
I think we have 2 options here:
- like this (I had to include `allowDangerousTypes=true` in Makefile CRD_OPTIONS to use float here)
- or in unsupportedOverride

since it is a new field, it needs to be documented, right? So I think either locations are ok

Reply:
if the only reason its dangerous is because of float, we could just use int and validate?

Reply:
did not understand the part about validate, but do not think we would lose much by being int here

another option is to be as in args, it is a string and is parsed as float

Reply:
Would resource.Quantity work here?

Reply:
I saw that in the issue I found the workaround with `allowDangerousTypes=true`, but did not test
going to try out

Reply:
from my tests, I would not use `resource.Quantity`

It is good for users (it can write `1000` or `1k`) but for us to read the value does not seem to be easy

From what I saw, we should use `String()` or `AsInt64()` method to get value. If I pass `1.5` I can not use `AsInt64()` to pass value to flag, and if I pass `1G` I can not use `String()` to pass value to flag (maybe I did not search enough)

If we do not want to use `allowDangerousTypes=true`, I think best option is to use string as in args https://github.com/openshift/oadp-operator/blob/145a6b82e4bec688294f90eba4a5ce2d96701752/pkg/velero/server/config.go#L58 (`int` we would be changing behavior of flag in Velero, so best to be cautious I think and convert string to float)

Reply:
I also see AsDec and AsApproximateFloat64, which might be more appropriate than AsInt64. But if it's troublesome then probably String is the easiest way to go. Thanks for trying it out.

Reply:
@shubham-pampattiwar comment "I am not so sure that we should add `allowDangerousTypes=true`"

Reply:
tried with controller-gen 0.14.0 but it also needs dangerous type. error message:

> found float, the usage of which is highly discouraged, as support for them varies across languages. Please consider serializing your float as string instead. If you are really sure you want to use them, re-run with crd:allowDangerousTypes=true

Reply:
I think using int is fine for OADP. We convert to string to pass to velero -- the fact that velero uses float isn't really relevant. I can't see any  reason to set QPS to 100.5 instead of 100.

Reply:
+1

Reply:
> it's technically supported, but floats are highly discouraged (and disallowed in official kubernetes APIs) because they won't round-trip properly, can't actually represent certain numbers exactly.

https://github.com/kubernetes-sigs/controller-tools/issues/245#:~:text=it%27s%20technically%20supported%2C%20but%20floats%20are%20highly%20discouraged%20(and%20disallowed%20in%20official%20kubernetes%20APIs)%20because%20they%20won%27t%20round%2Dtrip%20properly%2C%20can%27t%20actually%20represent%20certain%20numbers%20exactly

Reply:
as discussed in scrum, changed qps to int and removed dangerous type